Enclave/Enclave/Enclave.cpp

```diff
@@ -0,0 +1,13 @@
+#include "sgx_trts.h"
+#include <string.h>
+#include "ErrorCode.h"
+#include "Enclave_t.h"
+
+unsigned int e_generate_random_long(unsigned long *result) {
+	unsigned long longRand;
+	if (sgx_read_rand((unsigned char *)&longRand, 4) != SGX_SUCCESS) return FC_ERR_RAND;
```

Comment:
Client code expects random long. We need to convert `unsigned char*`s eventually and I prefer to do that in enclave since that way we can prove how final value was generated using SGX Attestation.

Fair point with sizeof(long) .